pkg/cluster/arooperator.go

```diff
@@ -36,16 +36,15 @@ func (m *manager) aroDeploymentReady(ctx context.Context) (bool, error) {
 	return m.aroOperatorDeployer.IsReady(ctx)
 }
 
-func (m *manager) ensureAROOperatorRunningDesiredVersion(ctx context.Context) error {
+func (m *manager) ensureAROOperatorRunningDesiredVersion(ctx context.Context) (bool, error) {
 	if !m.isIngressProfileAvailable() {
 		m.log.Error("skip ensureAROOperatorRunningDesiredVersion")
-		return nil
+		// skip and don't retry
+		return true, nil
 	}
-
-	err := m.aroOperatorDeployer.IsRunningDesiredVersion(ctx)
-	if err != nil {
-		m.log.Errorf("cannot ensureAROOperatorRunningDesiredVersion.IsRunningDesiredVersion: %s", err.Error())
```

Comment:
With the current implementation, this error message is not logged anymore but is possibly needed. For arooperatorReady and apiServerReady, we do not log the error in RP.